native/src/modules/endpoint/hocs/withPayloadProvider.js

```diff
@@ -60,11 +60,18 @@ const withPayloadProvider = <S: { dispatch: Dispatch<StoreActionType> }, R: {}>
   return (Component: React.ComponentType<S>): React.ComponentType<PropsType<S, R>> => {
     return class extends React.Component<PropsType<S, R>, {| timeoutExpired: boolean |}> {
       static displayName = wrapDisplayName(Component, 'withPayloadProvider')
+      timeout = null
 
       state = { timeoutExpired: false }
 
       componentDidMount () {
-        setTimeout(() => this.setState({ timeoutExpired: true }), LOADING_TIMEOUT)
+        this.timeout = setTimeout(() => this.setState({ timeoutExpired: true }), LOADING_TIMEOUT)
```

Comment:
I propose to use a hook here. It offers functionality for cleaning up: 

```
  useEffect(() => {
    const timer = setTimeout(() => {
      setCount('Timeout called!');
    }, 1000);
    return () => clearTimeout(timer);
  }, []);
```

Reply:
I kind of was expecting this :D However, to use a hook, we'd have to convert this to a functional component which I was currently not willing to do as I think this would have needed quite a lot of other changes.

Reply:
But definitely, with a hook it would be way cleaner :+1:

Reply:
That is a task of 3 minutes usually :) I think we should use new framework features if they make sense.

Reply:
Class Components are not used anymore in modern react